### DISCUSSION TITLE
Return current state from reducer

### DISCUSSION THREAD

**Discussion by moffpage** [badoo/MVIKotlin#221] [Q&A] [answered]
Should reducer return a new state in every situation? Example:
I have a model:

```
data class Model(
    val state: State
)
```

and I have a result

```
sealed class Result {
    data class ModelDidChange(val newState: State) : Result()
    data class ModelChangeFailed(val error: StateChangeError) : Result()
}
```
object ReducerImpl : Reducer<State, Result> {
    override fun State.reduce(result: Result): State = when (result) {
        is Result.AppealStatusDidChange -> State(appealStatus = result.newStatus)
        is Result.AppealStatusChangedFailed -> TODO() // return old state and handle error
    }
}
```

What should I return in the second case? Should I return the same copy? How should I do that, if that's the case? Cause I have no access to Store object in the Reducer.

**Comment by arkivanov** [comment, marked as answer]
You have the current state in the `Reducer` as receiver parameter, so you can use `copy` (to create a shallow copy) or `this`.

**Comment by moffpage** [comment]
Yeah… For some reason (it was probably late) I was dumb and something didn’t work. Anyway, thanks for help!
